dynamiq/nodes/agents/base.py

```diff
@@ -208,6 +208,72 @@ def validate_source(cls, v):
             raise ValueError(f"source must be a string, got {type(v).__name__}: {v}")
         return v
 
+    @model_serializer
+    def serialize_content(self):
+        """Serialize content dict, converting any BytesIO objects to base64 strings while preserving key structure."""
+
+        if self.content is None or not isinstance(self.content, dict):
+            return {"content": self.content, "source": self.source, "step": self.step}
+
+        serialized_content = {}
+
+        for key, value in self.content.items():
+
+            if isinstance(value, io.BytesIO):
+                current_pos = value.tell()
+                value.seek(0)
+                content_bytes = value.read()
+                value.seek(current_pos)
+
+                import base64
+
+                encoded = base64.b64encode(content_bytes).decode("utf-8")
+
+                file_info = {
+                    "content": encoded,
+                    "name": getattr(value, "name", f"file_{key}"),
+                    "content_type": getattr(value, "content_type", "unknown"),
+                    "description": getattr(value, "description", ""),
+                    "original_size": len(content_bytes),
+                }
+                serialized_content[key] = file_info
+
+            elif isinstance(value, list):
```

Comment:
Please use recursion to handle list/dict items to be sure that we handle the full strcuture and avoid duplication